app/models/user.rb

```diff
@@ -0,0 +1,33 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+
+  property :id,               Serial
+  property :name,             String
+  property :username,         String, :required => true, :unique => true, :messages =>
+                                    { :presence => 'Please enter a username',
+                                      :is_unique => 'We already have that username' }
+  property :email,            String, :required => true, :unique => true, :format => :email_address,
```

Comment:
Metrics/LineLength: Line is too long. [100/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)